File: tests/runtime/results/intermediate.cs
namespace IntermediateWorld.wit.Exports.test.results
{
    public class TestExportsImpl : ITestExports
    {
        public static float StringError(float a)
        {
            return Imports.test.results.ITestImports.StringError(a);
        }

        public static float EnumError(float a)
        {
            try {
                return Imports.test.results.ITestImports.EnumError(a);
            } catch (WitException<Imports.test.results.ITestImports.E> e) {
                throw new WitException(e.TypedValue, 0);
            }
        }

        public static float RecordError(float a)
        {
            try {
                return Imports.test.results.ITestImports.RecordError(a);
            } catch (WitException<Imports.test.results.ITestImports.E2> e) {
                throw new WitException(new ITestExports.E2(e.TypedValue.line, e.TypedValue.column), 0);
            }
        }

        public static float VariantError(float a)
        {
            try {
                return Imports.test.results.ITestImports.VariantError(a);
            } catch (WitException<Imports.test.results.ITestImports.E3> e)
                when (e.TypedValue.Tag == Imports.test.results.ITestImports.E3.Tags.E1) {
                    throw new WitException(ITestExports.E3.E1((ITestExports.E)Enum.Parse(typeof(ITestExports.E), e.TypedValue.AsE1.ToString())), 0);
            } catch (WitException<Imports.test.results.ITestImports.E3> e)
                when (e.TypedValue.Tag == Imports.test.results.ITestImports.E3.Tags.E2) {
                    throw new WitException(ITestExports.E3.E2(new ITestExports.E2(e.TypedValue.AsE2.line, e.TypedValue.AsE2.column)), 0);
            }
            catch {
                throw new Exception("unreachable");
            }
        }

        public static uint EmptyError(uint a)
        {
            return Imports.test.results.ITestImports.EmptyError(a);
        }

        public static void DoubleError(uint a)
        {
            Imports.test.results.ITestImports.DoubleError(a);
        }
    }
}

File: tests/runtime/results/intermediate.go
package export_test_results_test

import (
	imports "wit_component/test_results_test"

	. "go.bytecodealliance.org/pkg/wit/types"
)

func StringError(x float32) Result[float32, string] {
	return imports.StringError(x)
}

func EnumError(x float32) Result[float32, imports.E] {
	return imports.EnumError(x)
}

func RecordError(x float32) Result[float32, imports.E2] {
	return imports.RecordError(x)
}

func VariantError(x float32) Result[float32, imports.E3] {
	return imports.VariantError(x)
}

func EmptyError(x uint32) Result[uint32, Unit] {
	return imports.EmptyError(x)
}

func DoubleError(x uint32) Result[Result[Unit, string], string] {
	return imports.DoubleError(x)
}
